rest/data/aliases.go

```diff
@@ -9,14 +9,31 @@ import (
 	"github.com/pkg/errors"
 )
 
-// FindProjectAliases queries the database to find all aliases.
-// If the repoId is given, we default to repo aliases if there are no project aliases.
-// If aliasesToAdd are given, then we fold those aliases in and remove any that are marked as deleted.
-// If includeProjectConfig, a merged list of aliases defined on the project page and the project config YAML will be returned,
-// with aliases set on the project page taking precedence.
-func FindProjectAliases(projectId, repoId string, aliasesToAdd []restModel.APIProjectAlias, includeProjectConfig bool) ([]restModel.APIProjectAlias, error) {
-	var err error
-	var aliases model.ProjectAliases
+// FindMergedProjectAliases returns a merged list of aliases, with the order of precedence being:
+// 1. aliases defined on the project page
+// 2. aliases defined on the repo page
+// 3. aliases defined in the project config YAML
+// The includeProjectConfig flag determines whether to include aliases defined in the project config YAML.
+// If the aliasesToAdd parameter is defined, we fold those aliases in and remove any that are marked as deleted.
+func FindMergedProjectAliases(projectId, repoId string, aliasesToAdd []restModel.APIProjectAlias, includeProjectConfig bool) ([]restModel.APIProjectAlias, error) {
+	projectRef, err := model.FindBranchProjectRef(projectId)
```

Comment:
I think we want MergedProjectRef since we pass it into the construct function and check version control

Reply:
Right, because `VersionControlEnabled` can be overridden at the repo level as well. Thanks